acme/challengeProvider.go

```diff
@@ -2,55 +2,101 @@ package acme
 
 import (
 	"crypto/tls"
+	"strings"
 	"sync"
 
-	"crypto/x509"
+	"fmt"
+	"github.com/cenk/backoff"
+	"github.com/containous/traefik/cluster"
+	"github.com/containous/traefik/log"
 	"github.com/xenolf/lego/acme"
+	"time"
 )
 
-type wrapperChallengeProvider struct {
-	challengeCerts map[string]*tls.Certificate
-	lock           sync.RWMutex
+var _ acme.ChallengeProviderTimeout = (*challengeProvider)(nil)
+
+type challengeProvider struct {
+	store cluster.Store
+	lock  sync.RWMutex
 }
 
-func newWrapperChallengeProvider() *wrapperChallengeProvider {
-	return &wrapperChallengeProvider{
-		challengeCerts: map[string]*tls.Certificate{},
+func newMemoryChallengeProvider(store cluster.Store) *challengeProvider {
```

Comment:
This method adds no real value over just using `&challengeProvider{…}` in the code :angel: